ksqldb-common/src/main/java/io/confluent/ksql/serde/SerdeFeature.java

```diff
@@ -24,6 +24,15 @@
  */
 public enum SerdeFeature {
 
+  /**
+   * The format supports interaction with the Confluent Schema Registry.
+   *
+   * <p>Indicates whether or not a format can support CREATE statements that
+   * omit the table elements and instead determine the schema from a Confluent Schema Registry
+   * query.
+   */
+  SCHEMA_INFERENCE,
```

Comment:
💯

Reply:
We could even use this to get rid of `JSON_SR`, or the very least the `JsonSchemaFormat` class.

We could choose to get rid of `JSON_SR` by adding a `WITH` property that can be used with the `JSON` format that indicates it should use the SR, e.g. 

```sql
CREATE STREAM FOO WITH (format='json', use_schema_registry=true);
```

Even if we don't do that, we can internally translate any use of `JSON_SR` through to using the `JsonFormat` class with the `SCHEMA_INFERENCE` feature set.

Thoughts?

Reply:
We had discussed something like that with @derekjn and @MichaelDrogalis and decided that it would be the lesser of two evils (w.r.t. user confusion) if we just kept them as two separate formats to make it extra "clear" what the difference is. I'm always happy reducing the code overhead though if we can use it to get rid of `JsonSchemaFormat` class